cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/profile/ProfileDataset.java

```diff
@@ -138,28 +157,28 @@ public void createIfNotExists(ProfileId profileId, Profile profile) {
    * @throws ProfileConflictException if the profile is enabled
    */
   public void deleteProfile(ProfileId profileId) throws NotFoundException, ProfileConflictException {
-    MDSKey rowKey = getRowKey(profileId);
+    MDSKey rowKey = getProfileRowKey(profileId);
     Profile value = table.get(rowKey, Profile.class);
     if (value == null) {
       throw new NotFoundException(profileId);
     }
-    // TODO: https://issues.cask.co/browse/CDAP-13514, check for associations of the profile before deleting
     if (value.getStatus() == ProfileStatus.ENABLED) {
       throw new ProfileConflictException(
         String.format("Profile %s in namespace %s is currently enabled. A profile can " +
                         "only be deleted if it is disabled", profileId.getProfile(), profileId.getNamespace()),
         profileId);
     }
-    table.delete(getRowKey(profileId));
+    table.delete(getProfileRowKey(profileId));
   }
 
   /**
    * Delete all profiles in a given namespace.
    *
    * @param namespaceId the id of the namespace
    */
+  @VisibleForTesting
   public void deleteAllProfiles(NamespaceId namespaceId) {
```

Comment:
if this is only for testing, can it be package private?

Reply:
This is used by ProfileService, and several tests are using the deleteAll method in ProfileService to clean up profile, so I cannot make this package privatge